bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/AbstractTableViewer.java

```diff
@@ -779,6 +810,32 @@ private void internalRemove(final Object[] elements) {
 		}
 	}
 
+	/**
+	 * Returns the data of the last item on the viewer.
+	 *
+	 * @return may return null
+	 */
+	Object getLastElement() {
```

Comment:
Can we please have a test that this method does not trigger a lazy provider to load *ALL* data?

Assume a LazyProvider that is backed by a database/RestAPI/whatever and incrementally loads items as they are scrolled into the view, requesting the **last** might trigger loading the full dataset just to find out it is not an expandable node...

Reply:
setDisplayIncrementally(int incrementSize) is not guaranteed to work with {@link SWT#VIRTUAL} viewers. API doc is updated.